### ISSUE TITLE
chat focus loss when room owner re-assignes sides

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#2817]
Steps:
* create a game on MP server 
* join the game with another client (second client)
* start typing anything in game chat with the second client (not room owner)
* let first client (room owner) re-assign a side. Like choosing between "Network Player" and "Empty" and player names

What you'll observe at this point: chat focus on the second client gets lost. Focus goes elsewhere right while you typed something. (Thus breaking your typing.)

**Comment by Vultraz** [comment]
Honestly, this is expected. Most apps would do that.

**Comment by CelticMinstrel** [comment]
Stealing focus is never expected and is always bad. The only reason most apps do it is because it's not actually an easy problem to solve. For this particular case it's about a text field focus, so... how do you determine whether the user is currently typing into the text field? How do you ensure that the text field remains focused? It may be inevitable that events temporarily remove the focus, so what you'd have to do is ensure it gets restored before keyboard event handling continues.

I think this is a good thing to fix, eventually.

**Comment by CelticMinstrel** [comment]
Yeah, it shouldn't have to, so if it can be fixed in a way that doesn't remove the focus in the first place, that's even better.

**Comment by ProditorMagnus** [comment]
Also happens when someone joins or leaves game.

**Comment by jostephd** [comment]
This fixes it, but makes it so it's not possible to use up arrow / down arrow to scroll the sides list. It also makes it so the chat box is focused when that dialog is entered.

```
diff --git a/src/gui/dialogs/multiplayer/mp_join_game.cpp b/src/gui/dialogs/multiplayer/mp_join_game.cpp
index 760f9b9eb23..e36c296ff97 100644
--- a/src/gui/dialogs/multiplayer/mp_join_game.cpp
+++ b/src/gui/dialogs/multiplayer/mp_join_game.cpp
@@ -339,7 +339,7 @@ void mp_join_game::generate_side_list(window& window)
 
 	tree_view& tree = find_widget<tree_view>(&window, "side_list", false);
 
-	window.keyboard_capture(&tree);
+	window.keyboard_capture(&find_widget<chatbox>(&window, "chat", false));
 
 	tree.clear();
 	team_tree_map_.clear();
```

**Comment by vgaming** [comment]
I think that's pretty good. I am curious of course if we can insert some additional easy (yet maintainable) hack to detect up/down keystrokes and do the appropriate action, but even in the proposed form, I think it's worth it.

I would merge.

**Comment by soliton-** [comment]
Fixed with #5252.
